src/utils.ts

```diff
@@ -104,12 +104,42 @@ export async function findNode(cwd: string): Promise<string> {
   // Stage 3: If we still haven't found Node.js, try to find it via a subprocess.
   // This evaluates shell rc/profile files and makes nvm work.
   node ??= await findNodeViaShell(cwd)
+
+  // If volta isn't installed in a Volta folder, this test will fail.
+  // If anyone got a better idea for checking volta's presence, please let me know.
+  const voltaRegex = /Volta/i
+
+  // Stage 4: We have found Node.js, but it might be managed by Volta.
+  // This attempt to ask volta for the path to the node executable.
+  if (node && voltaRegex.test(node))
+    node = await findNodeUsingVoltaOnWindows(cwd) ?? node
+
   if (!node)
     throw new Error(`Unable to find 'node' executable.\nMake sure to have Node.js installed and available in your PATH.\nCurrent PATH: '${process.env.PATH}'.`)
   pathToNodeJS = node
   return node
 }
 
+function findNodeUsingVoltaOnWindows(cwd: string): Promise<string | undefined> {
+  if (process.platform !== 'win32')
+    return Promise.resolve(undefined)
+  return new Promise<string | undefined>((resolve) => {
+    const childProcess = spawn(`volta which node`, {
```

Comment:
instead of "volta", we should probably pass down the found executable, no? Or is it a different one?

Reply:
The executable we have is node, It's a different one.